lib/ShadowEffect.vala

```diff
@@ -184,4 +175,39 @@ public class Gala.ShadowEffect : Clutter.Effect {
 
         return true;
     }
+
+    private static void increment_shadow_users (string key) {
+        var shadow = shadow_cache.@get (key);
+
+        if (shadow == null) {
+            return;
+        }
+
+        shadow.users++;
+
+        uint timeout_id;
+        if (shadows_marked_for_dropping.unset (key, out timeout_id)) {
+            Source.remove (timeout_id);
+        }
+    }
+
+    private static void decrement_shadow_users (string key) {
+        var shadow = shadow_cache.@get (key);
+
+        if (shadow == null) {
+            return;
+        }
+
+        if (--shadow.users == 0) {
+            queue_shadow_drop (key);
+        }
+    }
+
+    private static void queue_shadow_drop (string key) {
+        shadows_marked_for_dropping[key] = Timeout.add_seconds (5, () => {
```

Comment:
```suggestion
        shadows_marked_for_dropping[key] = Timeout.add_seconds_once (5, () => {
```

Reply:
The *_once functions unfortunately do not ref the data and since vala unrefs it and it's also not used anywhere else, it gets immediately freed. This causes a crash when timeout runs.